dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+                                                                   Relationship relationship,
+                                                                   boolean enableVirtualMetadata)
+        throws SQLException {
+        List<RelationshipMetadataValue> resultingMetadataValueList = new LinkedList<>();
+        RelationshipType relationshipType = relationship.getRelationshipType();
+        HashMap<String, VirtualMetadataConfiguration> hashMaps;
+        String relationName;
+        Item otherItem;
+        int place = 0;
+        boolean isLeftwards;
+        if (StringUtils.equals(relationshipType.getLeftType().getLabel(), entityType)) {
+            hashMaps = virtualMetadataPopulator.getMap().get(relationshipType.getLeftwardType());
+            otherItem = relationship.getRightItem();
+            relationName = relationship.getRelationshipType().getLeftwardType();
+            place = relationship.getLeftPlace();
+            isLeftwards = false; //if the current item is stored on the left,
+            // the name variant is retrieved from the rightwards label
+        } else if (StringUtils.equals(relationshipType.getRightType().getLabel(), entityType)) {
+            hashMaps = virtualMetadataPopulator.getMap().get(relationshipType.getRightwardType());
+            otherItem = relationship.getLeftItem();
+            relationName = relationship.getRelationshipType().getRightwardType();
+            place = relationship.getRightPlace();
+            isLeftwards = true; //if the current item is stored on the right,
+            // the name variant is retrieved from the leftwards label
+        } else {
+            //No virtual metadata can be created
+            return resultingMetadataValueList;
+        }
+
+        if (hashMaps != null && enableVirtualMetadata) {
+            resultingMetadataValueList.addAll(handleRelationshipTypeMetadataMapping(context, item, hashMaps,
+                                                                                    otherItem, relationName,
+                                                                                    relationship, place, isLeftwards));
+        }
+        RelationshipMetadataValue relationMetadataFromOtherItem =
+            getRelationMetadataFromOtherItem(context, otherItem, relationName, relationship.getID(), place);
+        if (relationMetadataFromOtherItem != null) {
+            resultingMetadataValueList.add(relationMetadataFromOtherItem);
+        }
+        return resultingMetadataValueList;
+    }
+
+    //This method will retrieve a list of RelationshipMetadataValue objects based on the config passed along in the
+    //hashmaps parameter. The beans will be used to retrieve the values for the RelationshipMetadataValue objects
+    //and the keys of the hashmap will be used to construct the RelationshipMetadataValue object.
+    private List<RelationshipMetadataValue> handleRelationshipTypeMetadataMapping(Context context, Item item,
```

Comment:
Another `handle` method which looks more like a `find`, as it's returning a list of things it gathered.

Reply:
I've renamed it to findVirtualMetadataFromConfiguration
Combined with the JavaDocs I hope this clarifies the method's purpose